src/Microsoft.AspNet.Mvc.DataAnnotations/DataAnnotationsModelValidator.cs

```diff
@@ -71,5 +87,28 @@ public IEnumerable<ModelValidationResult> Validate(ModelValidationContext valida
 
             return Enumerable.Empty<ModelValidationResult>();
         }
+
+        private string GetErrorMessage(ModelValidationContextBase validationContext)
+        {
+            if (validationContext == null)
```

Comment:
These validations are coming too late. Do in the `public` methods, not a `private` method that happens to call.